apollo-federation/src/operation/mod.rs

```diff
@@ -66,7 +67,7 @@ static NEXT_ID: atomic::AtomicUsize = atomic::AtomicUsize::new(1);
 ///
 /// Note that we shouldn't add `derive(Serialize, Deserialize)` to this without changing the types
 /// to be something like UUIDs.
-#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Serialize)]
```

Comment:
@TylerBloom @lennyburdette @dariuszkuc Does the comment above this need to be updated after this change?  (The comment says not to add `Serialize` here, but adds `Serialize`? 😄

Reply:
Ah, I definitely should have made a comment about this during review. That's my bad. My thought was that serialize was ok but deserialize was not (and both was definitely not a good idea); however, this should probably be feature-gated like the `snapshot!` macro that was added. I need to do some clean up around the `Serialize` implementation of a few types. I can bundle this change into that PR.

Reply:
Thank you! that'd be wonderful. 🙇  Consider my curiousity placated. 😄

Reply:
Addressed in this PR: https://github.com/apollographql/router/pull/5724